### PR TITLE
Draft: Repl Server documentation bug fixes

### PR DESCRIPTION
**Pull request by dhoomakethu**

### CODE REVIEW

**Comment by janiversen** [comment]
Nice to see a PR from you. Having a break from your work ?

As you might have seen I am working on refactoring the clients, and so far I have managed to make a much simpler structure, and a documentation that is understandable with examples embedded.

**Comment by dhoomakethu** [comment]
@janiversen this is ready to be merged, please merge it when you get time. Don't want to step in to your refactoring work.
Thanks.

**Comment by janiversen** [comment]
Will do, starting to rebase on current dev (seems yours is a bit old).

Seems you have a pylint problem, will look at that.

**Comment by dhoomakethu** [comment]
> Will do, starting to rebase on current dev (seems yours is a bit old).
> 
> Seems you have a pylint problem, will look at that.

I merged the latest dev on to this branch., JFYI

**Comment by janiversen** [comment]
This was a larger edit, there was a class half finished (only definition) and quite a number of places I had to correct, so please look at the result and make corrections if I did it wrongly.

**Comment by janiversen** [comment]
I will merge if I get green light.

**Review by janiversen — APPROVED** [review]
LGTM.

File: pymodbus/repl/client/main.py
Comment (dhoomakethu) [inline pymodbus/repl/client/main.py:44]: @janiversen  With raw string , the inline replacement of `{version}` does not work. we end up with literal `{version}` in the output instead of the actual output.  We need to revert back ti `f-strings` here

File: pymodbus/repl/client/mclient.py
Comment (dhoomakethu) [inline pymodbus/repl/client/mclient.py:75]: Broadcast to be assumed only when the unit is 0, this will assume a broadcast request when `unit` is not available in the `kwargs` and prints a wrong message instead of the actual exception.

File: pymodbus/repl/server/main.py
Comment (dhoomakethu) [inline pymodbus/repl/server/main.py:131]: Let's use explicit annotations from `typing` module , `list[int]` might result in `TypeError: 'type' object is not subscriptable`

File: pymodbus/repl/server/main.py
Comment (dhoomakethu) [inline pymodbus/repl/server/main.py:48]: Same here `list[str]` will raise TypeError, needs to be `typing.List`

**Comment by dhoomakethu** [comment]
@janiversen  Also,  This change results in `RuntimeError` 
```
/Users/sanjay/repo/pymodbus/pymodbus/server/async_io.py:485 in __init__                          │
│                                                                                                  │
│    482 │   │   │   │   │   │   │   │   │   │   response                                          │
│    483 │   │   """                                                                               │
│    484 │   │   self.active_connections = {}                                                      │
│ ❱  485 │   │   self.loop = asyncio.get_running_loop()                                            │
│    486 │   │   self.allow_reuse_address = allow_reuse_address                                    │
│    487 │   │   self.decoder = ServerDecoder()                                                    │
│    488 │   │   self.framer = framer or ModbusSocketFramer
RuntimeError: no running event loop
```
I had this change to address this particular issue, Looks like it got lost with your changes.

```
diff --git a/pymodbus/server/async_io.py b/pymodbus/server/async_io.py
index 58954ff9..92aed0d9 100755
--- a/pymodbus/server/async_io.py
+++ b/pymodbus/server/async_io.py
@@ -482,7 +482,7 @@ class ModbusTcpServer:  # pylint: disable=too-many-instance-attributes
                                         response
         """
         self.active_connections = {}
-        self.loop = asyncio.get_running_loop()
+        self.loop = kwargs.get("loop") or asyncio.get_event_loop()
         self.allow_reuse_address = allow_reuse_address
         self.decoder = ServerDecoder()
         self.framer = framer or ModbusSocketFramer
```

Let me know if I can revert these changes. Thanks.

**Comment by dhoomakethu** [comment]
To test the server you can run 
```
pip install -e .
pymodbus.server run --modbus-server tcp --framer socket --unit-id 1
```

File: pymodbus/repl/client/main.py
Comment (janiversen) [reply pymodbus/repl/client/main.py:44]: Ok, but then I need to add some pylint disable.

File: pymodbus/repl/server/main.py
Comment (janiversen) [reply pymodbus/repl/server/main.py:131]: List from typing is deprecated in 3.9, and the suggestion is to use list (builtin type).

The error you get is because I forgot an import at the top "from __future__ import annotations"

So let us not use List from the typing module, or we no longer support higher versions.

File: pymodbus/repl/server/main.py
Comment (janiversen) [reply pymodbus/repl/server/main.py:57]: Nope

**Comment by janiversen** [comment]
I have added your patch around get_event_loop(), but honestly if you run an async application the loop should be running.

Problem is that many run a sync application and expect to be able to use the async module. This is working for now.

I have updated the code, I leave the testing to you.

File: pymodbus/repl/client/mclient.py
Comment (janiversen) [reply pymodbus/repl/client/mclient.py:75]: the problem is that unit is an unused variable so I tried to get around that.

File: pymodbus/repl/server/main.py
Comment (janiversen) [reply pymodbus/repl/server/main.py:48]: nope

**Comment by dhoomakethu** [comment]
Thanks, now the repl client is broken, I will fix them and push it. Thanks

**Comment by janiversen** [comment]
Please note there have been a couple of changes in the clients. One of the more important ones is that you need to call client.start() to get a connection. Please see the updated doc.

**Review by janiversen — APPROVED** [review]

**Comment by janiversen** [comment]
Lets get this into dev, then we can both work in parallel again.

**Pull request merged**
